apps/web/src/app/[locale]/[username]/_components/dashboard/challenge-history.tsx

```diff
@@ -0,0 +1,46 @@
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '@repo/ui/components/table';
+import Link from 'next/link';
+import { getRelativeTime } from '~/utils/relativeTime';
+
+interface Challenge {
+  id: number;
+  name: string;
+  slug: string;
+  submission: {
+    createdAt: Date;
+  }[];
+}
+
+interface Props {
+  challenges: Challenge[];
+}
+
+export default function ChallengeHistory({ challenges }: Props) {
```

Comment:
lets use pagination. if you look at the datatables in the admin project under tracks you can copy pasta most of it